test drupal 10

byrond opened the pull request:
### Description

_A description of the changes. Consider including images/gifs to demonstrate behaviors (e.g. using [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture))._

### Testing instructions

1. `composer create-project palantirnet/drupal-skeleton drupal10 dev-drupal-10 --no-interaction`
2. `cd drupal10`
3. Edit `.ddev/config.yaml` and change the `name` to `drupal10`
4. `ddev start`
5. `ddev ssh`
6. `vendor/bin/the-build-installer`
7. Visit https://drupal10.ddev.site/ and verify the site is running.
8. Log in, and verify the site is running Drupal 10.

### Remaining tasks

- [x] Review and merge https://github.com/palantirnet/the-build/pull/201
- [x] Create a new release of `the-build`
- [x] Change line 30 here: https://github.com/palantirnet/drupal-skeleton/pull/126/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R30 to use `^4@beta` (or `^4` if there is a stable release).

byrond committed:
test drupal 10

byrond committed:
use php 8.1 on circle

byrond committed:
use "lenient" drupal repo

byrond committed:
fix composer repositories json

byrond committed:
update drupal core-dev version

byrond committed:
add composer-drupal-lenient

byrond committed:
allow the composer-drupal-lenient plugin

byrond committed:
use the-build drupal-10 branch

byrond committed:
remove incompatible drupal-extension

byrond committed:
update to composer/installers 2.x

byrond committed:
remove drupal contrib modules for now

byrond committed:
use minimum-stability dev

byrond committed:
add phing-composer-configurator to allowed plugins

byrond committed:
add phpstan extension-installer to allowed plugins

byrond committed:
revert changes that are not core/module updates

byrond committed:
Merge remote-tracking branch 'origin/develop' into drupal-10

byrond committed:
use alpha release for drupal-extension

byrond committed:
test the skeleton with the d10 utility module update

byrond committed:
add mink-goutte-driver as a dependency since it was removed from Drupal core

byrond committed:
change DDEV config type to drupal10

becw commented on composer.json:
```diff
@@ -15,19 +15,20 @@
-        "drupal/workbench_tabs": "^1.5"
+        "composer/installers": "^2.0",
+        "drupal/admin_toolbar": "^3",
+        "drupal/config_split": "^1",
```
I had to use config_split ^2 in another Drupal 10 project I set up. Not sure if we'll need to do that here as well.

byrond replied:
1.9 is compatible with Drupal 10, and 2.0.0 is still in pre-release, so I think we can leave this for now.

byrond committed:
Merge branch 'develop' into drupal-10

byrond committed:
use version 4-beta of the-build

byrond committed:
add comma lost during merge

byrond committed:
add drupal9 instructions to readme

byrond committed:
update full project setup section with d9 instructions

byrond committed:
Merge branch 'develop' into drupal-10

byrond committed:
remove config_ignore, because there is no release for D10 yet

Kbentham commented:
I am seeing some errors that I think are related to this issue on the module and status pages https://www.drupal.org/project/drupal/issues/3293463

https://drupal10.ddev.site/admin/modules
<img width="1394" alt="Screen Shot 2023-01-11 at 1 47 00 PM" src="https://user-images.githubusercontent.com/1236362/211903336-78a91234-502a-4850-a439-d2ff806f0b77.png">

https://drupal10.ddev.site/admin/reports/status
<img width="1591" alt="Screen Shot 2023-01-11 at 1 47 24 PM" src="https://user-images.githubusercontent.com/1236362/211903354-97ea4ccd-7d45-4bb3-896f-fe6666b32f83.png">
 other than that this is looking good!

byrond committed:
add patch for drupal core issue 3293463

byrond committed:
Revert "add patch for drupal core issue 3293463"

This reverts commit ed574e7d44a616842bdf27766d6df7a3678c0e2d.

byrond commented:
@Kbentham I added the patch in the Drupal core issue you found, but I still got the same error after first install and going to the status page. Clearing cache made the error go away. I removed the patch, so could you confirm that clearing cache after installing the site allows the status page to work?

byrond commented:
We might need to update The Build to clear cache when it's done switching the profile, so this manual step isn't required after running the first Drupal install.

Kbentham reviewed (approved):
Looks good to me
